compiler/src/iree/compiler/PluginAPI/PluginManager.h

```diff
@@ -108,6 +108,18 @@ class PluginManagerSession : public PipelineExtensions {
     }
   }
 
+  std::string detectCustomInputOperations(ModuleOp &module) override {
+    for (auto *s : initializedSessions) {
+      // Return the type mnemonic from the first session that detects support.
+      std::string typeMnemonic = s->detectCustomInputOperations(module);
+      if (!typeMnemonic.empty()) {
+        // TODO(scotttodd): error if multiple sessions found ops they support?
```

Comment:
Ideally we can support multiple of these together, could end up populating many pipelines together (e.g., why not some TOSA, StableHLO and ATen ops all together?:)

Reply:
Yeah, agreed. I think putting the input conversions on equal footing as plugins (with a clear API to implement) will also help untangle some of those design questions. As it is now, we have a mix of conversions following different sets of rules.